lib/puppet/functions/simplib/validate_port.rb

```diff
@@ -0,0 +1,39 @@
+# Validates whether or not the passed argument is a valid port
+# (i.e. between `1` - `65535`).
+Puppet::Functions.create_function(:'simplib::validate_port') do
```

Comment:
@lnemsick-simp Also, as @jeefberkey indicated in his first post, using `assert_type` should also suffice for most of our `validate` functions.